contracts/HypNative.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {TokenRouter} from "./libs/TokenRouter.sol";
+import {Message} from "./libs/Message.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+/**
+ * @title Hyperlane Native Token Router that extends ERC20 with remote transfer functionality.
+ * @author Abacus Works
+ * @dev Supply on each chain is not constant but the aggregate supply across all chains is.
+ */
+contract HypNative is TokenRouter {
+    /**
+     * @notice Constructor
+     * @param gasAmount Amount of destination gas to be paid for processing
+     */
+    constructor(uint256 gasAmount) TokenRouter(gasAmount) {}
+
+    /**
+     * @notice Initializes the Hyperlane router, ERC20 metadata, and mints initial supply to deployer.
+     * @param _mailbox The address of the mailbox contract.
+     * @param _interchainGasPaymaster The address of the interchain gas paymaster contract.
+     */
+    function initialize(address _mailbox, address _interchainGasPaymaster)
+        external
+        initializer
+    {
+        // transfers ownership to `msg.sender`
+        __HyperlaneConnectionClient_initialize(
+            _mailbox,
+            _interchainGasPaymaster
+        );
+    }
+
+    /**
+     * @inheritdoc TokenRouter
+     * @dev uses (`msg.value` - `_amount`) as interchain gas payment and `msg.sender` as refund address.
+     */
+    function transferRemote(
+        uint32 _destination,
+        bytes32 _recipient,
+        uint256 _amount
+    ) external payable override {
+        require(msg.value >= _amount, "msg.value < amount");
+        uint256 gasPayment = msg.value - _amount;
+        _dispatchWithGas(
```

Comment:
Should return message ID

Reply:
fixed in https://github.com/hyperlane-xyz/hyperlane-token/pull/44